packages/components/src/drop-zone/provider.js

```diff
@@ -233,13 +233,13 @@ class DropZoneProvider extends Component {
 	}
 
 	render() {
-		const { children } = this.props;
-		return children;
+		return (
+			<Provider value={ this.dropZoneCallbacks }>
+				{ this.props.children }
+			</Provider>
+		);
 	}
 }
```

Comment:
I'm having trouble following the changes here? I guess you refactored more than necessary to just support the new context API? any context on the changes?

Reply:
No functionality has been changed, but did some refactoring to -hopefully- improve the readibility of this big class (commits are atomic and may provide useful info). Here's a summary:

- renamed some variables for consistency and clarity (`dropZones` instead of `dropzones`, `onDragOver` instead of `dragOverListener`, etc).
- extracted some methods out of the class (`isWithinZoneBounds`, `isTypupported`, `getDragEventType`).

These are not strictly necessary, I can revert them if they're blockers to understand this. Didn't want to create a separate PR to address those.